### DISCUSSION TITLE
How to change the color in vanilla js?

### DISCUSSION THREAD

**Discussion by thecco** [visgl/deck.gl#8236] [Q&A] [answered]
I am using the vanilla js and I cannot figure out for the life of me why the updateTriggers is not working. The documentation isn't giving me much help as I did exactly as it said.
I'm not a web developer, so I don't know js well.
```
var hoveredObject  = null;
const pathLayer = new PathLayer({
    id: 'path-layer',
    data: pipeData,
    pickable: true,
    getPath: d => d.path,
    getColor: d => d.properties.isHovered ? [0, 255, 0] :  [255, 0, 0] ,
    widthUnits: 'pixels',
    parameters: {
        depthMask: false
    },
    autoHighlight: false,
    updateTriggers: {
        getColor: { hoveredObject }
    },
    onHover: (info) => onNodeHover(info)
});
```

```
function onNodeHover(info) {
    if(hoveredObject){
        hoveredObject.properties.isHovered= false;
    }

    info.object.properties.isHovered = true;
    hoveredObject = info.object;
}
```

**Comment by ibgreen** [comment, marked as answer]
Just to make sure it is understood, you need to re-render your layer (i.e. pass a new copy to deck.setProps({layers: [pathLayer]}) for anything to happen. 

deck uses a diffing algorithm on props, it doesn't set up a watch on your variables.
